rx-lite/index.d.ts

```diff
@@ -232,7 +232,7 @@ declare namespace Rx {
         withLatestFrom<TOther, TResult>(souces: (Observable<TOther> | IPromise<TOther>)[], resultSelector: (firstValue: T, ...otherValues: TOther[]) => TResult): Observable<TResult>;
         concat(...sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
         concat(sources: (Observable<T> | IPromise<T>)[]): Observable<T>;
-        concatAll(): Observable<T>;
+        concatAll(): T;
```

Comment:
This method is different from other methods such as `map` or `filter` in that it reduces an observable of observables onto a flattened observable. It does not returns the same type as `x`.

Since the reduction is not recursive, if `x` is `Observable<Observable<Observable<T>>>` `concat` should be of type `Observable<Observable<T>>`. This is compatible with the pull request

If `x` is `Observable<T>` ideally `concatAll` should not be available since[ the method expects an observable of observables as an input](https://github.com/Reactive-Extensions/RxJS/blob/8fa95ac884181fb6cbff8ce7c1d669ffb190f5e4/src/modular/observable/mergeall.js#L60), but this kind of type restrictions [are not possible to express on typescript yet](https://github.com/Microsoft/TypeScript/issues/1290)